### PR TITLE
Fix union queries not working in @requires

### PR DESCRIPTION
**Pull request by rickbijkerk**
fixes: https://github.com/99designs/gqlgen/issues/3342

also tested in our project and its working ; )

I have:
 - [x ] Added tests covering the bug / feature (see [testing](https://github.com/99designs/gqlgen/blob/master/TESTING.md))
 - [x ] Updated any relevant documentation (see [docs](https://github.com/99designs/gqlgen/tree/master/docs/content))

### CODE REVIEW

File: plugin/federation/federation.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	_ "embed"
 	"errors"
 	"fmt"
+	"github.com/99designs/gqlgen/internal/rewrite"
```
Comment (StevenACoffman) [inline plugin/federation/federation.go:7]: Oh thanks!

Comment (rickbijkerk) [reply plugin/federation/federation.go:7]: ah must have been auto formatted, let me change that!

**Comment by clayne11** [comment]
`explicit_requires` is deprecated — did you test using `computed_requires` and does this issue still occur there?

**Comment by rickbijkerk** [comment]
yes, also tried it with `computed_requires` same issue occurs there

**Comment by clayne11** [comment]
Got it, thanks. I believe this code is not really used in the `computed_requires` flow but it is still executed until we fully remove the `explicit_requires` code path. Thanks for the fix.

**Comment by StevenACoffman** [comment]
@rickbijkerk Hey, can you run:
```
go generate ./...
```
Thanks for this!

**Pull request merged**